### PR TITLE
MAINT: use super() as described by PEP 3135

### PR DESCRIPTION
**Pull request by mwtoews**
There are two common ways to use the [`super` built-in](https://docs.python.org/3/library/functions.html#super):

1. Passing both the class and instance, e.g. `super(C, self)`, as was required for Python 2
2. Just using `super()` without arguments, as described by [PEP 3135](https://www.python.org/dev/peps/pep-3135/), adopted in 2007 for Python 3.0

Currently there is a mixture of the two forms; find them with `git grep "super([^)]"` or `git grep "super()"`. This PR refactors most of them to the PEP 3135 convention. This convention is cleaner and enforces the DRY (Don't Repeat Yourself) rule.

There are a few exceptions that should remain as they are (i.e. not touched by this PR):
1. [scipy/stats/_continuous_distns.py#L2788](https://github.com/scipy/scipy/blob/f679c65d641f6dbd5217516670ded71d60081076/scipy/stats/_continuous_distns.py#L2788)
2. [scipy/stats_distn_infrastructure.py#L3700](https://github.com/scipy/scipy/blob/f679c65d641f6dbd5217516670ded71d60081076/scipy/stats/_distn_infrastructure.py#L3700)

With both of these instances, `super(cls, self)`, the first argument *cls* is a different class to the parent class. There are a few other uses of `super()` used in methods outside classes that should also remain with arguments.

### CODE REVIEW

**Comment by tupui** [comment]
Sounds like a good think to modernise the code base! We could also take care of the `class ... (object):` afterward.

File: scipy/signal/ltisys.py [hunk 1]
```diff
@@ -1337,7 +1337,7 @@ def __init__(self, *system, **kwargs):
             return
 
         # Remove system arguments, not needed by parents anymore
-        super(StateSpace, self).__init__(**kwargs)
```
Comment (tupui) [inline scipy/signal/ltisys.py:1340]: Can it be that due to the possible inheritence change in `__new__` then the super here could call something different than `StateSpace`?

Same comment for the other `__init__` from this file.

**Review by tupui — COMMENTED** [review]
I checked at all the modifications and can confirm it just changes cases like these:

```python
class Foo(Bar):
    def __init__(self):
        # super(Foo, self).__init__(...)
        # changed to
        super().__init__(...)
```

I am just not sure about some specific cases with `__new__`.

**Comment by mwtoews** [comment]
> I am just not sure about some specific cases with `__new__`.

Good question. I'll need to dig deeper with a simple example. Most of these changes were "detected" using an [ast](https://docs.python.org/3/library/ast.html) script that finds calls to "super" with two arguments, where the first is the class and the second is the first argument from the method. It's possible this is a naïve approach.

**Comment by tupui** [comment]
Any update? If you don't have time, we could just remove the concerning parts for now.

**Comment by mwtoews** [comment]
From a few small examples that I was looking at, I didn't see any glaring differences of behavior with refactoring `super` in `__new__`.

But for the `__new__` instances in this PR, it would also be fine to keep them in the old style to be more explicit. These are the files:
- scipy/signal/ltisys.py
- scipy/sparse/linalg/interface.py
- scipy/stats/_distn_infrastructure.py

Here is my (hacky) [super_sniffer.py](https://gist.github.com/mwtoews/2da8d6e1d757a3fca26069d3ddb82577) script that I used to check which scenarios this refactoring is appropriate (or not). It does not consider handling `__new__` any differently, but will show the method where `super` is used.

**Comment by tupui** [comment]
Then I would keep the explicit super in cases we are not 100 sure for now (I don't have time to test this myself for now). This can be addressed later on.

**Comment by mwtoews** [comment]
The previous commit undoes simpler super calls in `__new__` methods for the following:

scipy/signal/ltisys.py

- line 54: LinearTimeInvariant
- line 222: lti
- line 406: dlti
- line 586: TransferFunction
- line 968: ZerosPolesGain
- line 1331: StateSpace

scipy/sparse/linalg/interface.py

- lines 145, 147: LinearOperator

scipy/stats/_distn_infrastructure.py

- lines 3024, 3027: rv_discrete

**Review by tupui — APPROVED** [review]

**Comment by tupui** [comment]
Thanks @mwtoews. I am merging (CI issue not related) this now to avoid another maintainer to have to go through this. Feel free to open a follow up PR to finish this.

**Pull request merged**